File: R5ProObjectiveCExamples/R5ProObjectiveCExamples/Tests/Subscribe/SubscribeTest.m
//
//  SubscribeTest.m
//  R5ProObjectiveCExamples
//
//  Created by David Heimann on 6/12/17.
//  Copyright © 2017 Infrared5. All rights reserved.
//

#import "SubscribeTest.h"
#import "Testbed.h"

@interface SubscribeTest ()

@property int current_rotation;

@end

@implementation SubscribeTest

-(void) viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    [self setupDefaultR5VideoViewController];
    
    R5Configuration* config = [self getConfig];
    R5Connection* connection = [[R5Connection alloc] initWithConfig:config];
    
    self.subscribeStream = [[R5Stream alloc] initWithConnection:connection];
    self.subscribeStream.delegate = self;
    self.subscribeStream.client = self;
    
    [self.currentView attachStream:self.subscribeStream];
}

-(void) updateOrientation:(int)value {
    
    if( _current_rotation == value ){
        return;
    }
    
    _current_rotation = value;
    self.currentView.view.layer.transform = CATransform3DMakeRotation((CGFloat)value, 0.0, 0.0, 0.0);
}

-(void) onMetaData:(NSString*)data {
    
    NSArray* props = [data componentsSeparatedByString:@";"];
    for (NSString* keyValue in props) {
        NSArray* kv = [keyValue componentsSeparatedByString:@"="];
        if( [kv[0] isEqualToString:@"orientation"] ){
            NSString* value = kv[1];
            [self updateOrientation: [value intValue] ];
        }
    }
}

@end

File: R5ProTestbed/Tests/Subscribe/SubscribeTest.swift
//
//  SubscribeTestViewController.swift
//  R5ProTestbed
//
//  Created by Andy Zupko on 12/16/15.
//  Copyright © 2015 Infrared5. All rights reserved.
//

import UIKit
import R5Streaming

@objc(SubscribeTest)
class SubscribeTest: BaseTest {
    
    var current_rotation = 0;

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        
        setupDefaultR5VideoViewController()
        
        let config = getConfig()
        // Set up the connection and stream
        let connection = R5Connection(config: config)
        self.subscribeStream = R5Stream(connection: connection)
        self.subscribeStream!.delegate = self
        self.subscribeStream?.client = self
        
        currentView?.attach(subscribeStream)
        
        
        self.subscribeStream!.play(Testbed.getParameter(param: "stream1") as! String)
        
        

    }
    
    func updateOrientation(value: Int) {
        
        if current_rotation == value {
            return
        }
        
        current_rotation = value
        currentView?.view.layer.transform = CATransform3DMakeRotation(CGFloat(value), 0.0, 0.0, 0.0);
        
    }
    
    func onMetaData(data : String) {
        
        let props = data.characters.split(separator: ";").map(String.init)
        props.forEach { (value: String) in
            let kv = value.characters.split(separator: "=").map(String.init)
            if (kv[0] == "orientation") {
                updateOrientation(value: Int(kv[1])!)
            }
        }
        
    }
    
}
